docs/core-services/crypto-service.md

```diff
@@ -7,18 +7,138 @@ The interface centralizes cryptographic operations and keystore password managem
+
+!!! warning
+
+    Failing to replace the default secret leaves encrypted data vulnerable. Ensure secrets are stored and managed securely in your environment.
+
+The secret key must be either 16, 24, or 32 bytes (characters) long, it can be specified in the following ways:
+
+### Using systemd-credentials (recommended)
+
 !!! note
-    The default `CryptoServiceImpl` bundled with Kura uses a well-known, default encryption key when no secret is provided. This default is intended only for development and testing purposes. For production deployments, users MUST replace the default secret by setting the Java environment variable:
+    systemd-credentials is available starting from systemd version 250. This method cannot be used on container based deployments.
 
-    - org.eclipse.kura.core.crypto.secretKey
+Strating from Eclipse Kura 6.0, the default `CryptoService` implementation supports loading the encryption key from a [systemd credential](https://systemd.io/CREDENTIALS/) named `kura_encryption_key`.
 
-    The secret key must be either 16, 24, or 32 bytes (characters) long. Example (set as JVM argument or environment variable depending on your runtime):
+It can be speficied using any of the methods supported by systemd. For example it can be stored in encrypted form in a configuration dropin of the `kura.service` unit, as shown below (adapted from Example 2 in [systemd-creds](https://www.freedesktop.org/software/systemd/man/latest/systemd-creds.html) man page).
 
-    - As JVM system property: -Dorg.eclipse.kura.core.crypto.secretKey=your-16+char-secret
+The commands should be run as root.
 
-    Failing to replace the default secret leaves encrypted data vulnerable. Ensure secrets are stored and managed securely in your environment.
+```sh
+mkdir -p /etc/systemd/system/kura.service.d
+```
+
+```sh
+systemd-ask-password -n | ( echo "[Service]" && systemd-creds encrypt --name=kura_encryption_key -p - - ) >/etc/systemd/system/kura.service.d/50-encryption-key.conf
+```
+
+```sh
+systemctl daemon-reload
+```
+
+The result can be verified with `systemctl cat kura`, it should produce an output similar to the following:
+
+```sh
+systemctl cat kura
+# /lib/systemd/system/kura.service
+[Unit]
+Description=Kura
+Wants=dbus.service
+After=dbus.service
+
+....
+kura.service content
+....
+
+# /etc/systemd/system/kura.service.d/50-encryption-key.conf
+[Service]
+SetCredentialEncrypted=kura_encryption_key: \
+        Whxqht+dQJax1aZeCGLxmiAAAAABAAAADAAAABAAAADQfAnaQJMAVKCEJjcAAAAASk3/B \
+        EZuKkHQPNKDXe7zn68bjyhzE7ni2R+g2B9o9aWrtMT9OGztsK+WbpsjTr8ci4FKcFL/dd \
+        B1nKZ+O2Zt1Q==
+
+```
+
+!!! note
+
+    The setup above is just an example, systemd offers different ways to pass credentials to services. Please review systemd documentation to understand the different options and chose the most appropriate for your case.
+
+### Using a Java system property
```

Comment:
To make the available alternatives stand out more I would add "Option X" in the paragraph title.



```suggestion
### Option 2: Using a Java system property
```